pkg/querier/tripperware/instantquery/instant_query.go

```diff
@@ -213,3 +227,22 @@ func decorateWithParamName(err error, field string) error {
 	}
 	return fmt.Errorf(errTmpl, field, err)
 }
+
+func marshalResponse(resp *tripperware.PrometheusResponse, acceptHeader string) (string, []byte, error) {
+	for _, clause := range goautoneg.ParseAccept(acceptHeader) {
+		if jsonMIMEType.Satisfies(clause) {
+			b, err := json.Marshal(resp)
+			return tripperware.ApplicationJson, b, err
+		} else if rulerMIMEType.Satisfies(clause) {
+			b, err := resp.Marshal()
+			return tripperware.QueryResponseCortexMIMEType, b, err
+		}
+	}
+
+	b, err := json.Marshal(resp)
+	return tripperware.ApplicationJson, b, err
+}
+
+func isSourceRuler(h *http.Header) bool {
```

Comment:
I like that we identified query traffic and ruler traffic now.
Can we also add this to `frontend/transport/handler.go`? We can have additional logging about the query source and update metrics (like queries from rule or API, etc).

Reply:
Do you mean to add labels to query stats metrics?
```
h.queryScannedSamples = promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
			Name: "cortex_query_samples_scanned_total",
			Help: "Number of samples scanned to execute a query.",
		}, []string{"user", "source"})
```

And add a log field to `ServeHTTP`. right?

Reply:
> Do you mean to add a label (source) to query stats metrics?

We don't need to add this for all metrics. Only the counter for number of requests should be good enough

Reply:
We don't have a counter metric in the handler.
Should be add it?
FYI. We have a `cortex_ruler_query_frontend_request_duration_seconds` histogram.

Reply:
I add a `User-Agent` field to logger.
`logger := util_log.WithUserAgent(userAgent, util_log.WithContext(r.Context(), f.log))`

Reply:
Umm, we actually already have one called `queriesPerTenant` but it is in `roundtrip.go`. Sorry for the wrong info.

Reply:
thanks!
I add a `source` label to `cortex_query_frontend_queries_total`.
The label value is `ruler` or 'others`.
WDYT?

Reply:
Can we call it `API` and `Rule`? This aligns with https://github.com/cortexproject/cortex/blob/master/pkg/cortexpb/cortex.proto#L14